src/devices/Tlc1543/Tlc1543.cs

```diff
@@ -0,0 +1,239 @@
+            _dataOut = dataOut;
+            _inputOutputClock = inputOutputClock;
+
+            _digital.OpenPin(_address, PinMode.Output);
+            _digital.OpenPin(_chipSelect, PinMode.Output);
+            _digital.OpenPin(_dataOut, PinMode.InputPullUp);
+            _digital.OpenPin(_inputOutputClock, PinMode.Output);
+        }
+
+        /// <summary>
+        /// Constructor for Tlc1543
+        /// </summary>
+        /// <param name="address">Address</param>
+        /// <param name="chipSelect">Chip Select</param>
+        /// <param name="dataOut">Data Out</param>
+        /// <param name="endOfConversion">End of Conversion</param>
+        /// <param name="inputOutputClock">I/O Clock</param>
+        public Tlc1543(int address, int chipSelect, int dataOut, int inputOutputClock, int endOfConversion)
+        {
+            _address = address;
+            _chipSelect = chipSelect;
+            _dataOut = dataOut;
+            _endOfConversion = endOfConversion;
+            _inputOutputClock = inputOutputClock;
+
+            _digital.OpenPin(_address, PinMode.Output);
+            _digital.OpenPin(_chipSelect, PinMode.Output);
+            _digital.OpenPin(_dataOut, PinMode.InputPullUp);
+            _digital.OpenPin((int)_endOfConversion, PinMode.InputPullUp);
+            _digital.OpenPin(_inputOutputClock, PinMode.Output);
+        }
+
+        #endregion
+
+        #region Methods
+
+        /// <summary>
+        /// Reads sensor value in Fast Mode 1 (10 clock transfer using !chipSelect)
+        /// <remarks>
+        /// <br>1 cycle used - gets data on second call of this method</br>
+        /// <br>First cycle: Ask for value on the channel</br>
+        /// </remarks>
+        /// </summary>
+        /// <param name="channelNumber">Channel to read</param>
+        /// <returns>10 bit value corresponding to relative voltage level on channel</returns>
+        public int ReadChannel(int channelNumber)
+        {
+            if (channelNumber < 0 & channelNumber > 14)
+            {
+                throw new ArgumentOutOfRangeException("channelNumber", "Out of range. \nMust be non-negative and less than 14. \nRefer to table 2 and 3 in TLC1543 documentation for more information");
+            }
+
+            int value = 0;
+            _digital.Write(_chipSelect, 0);
+            for (int i = 0; i < 10; i++)
+            {
+                if (i < 4)
+                {
+                    // send 4-bit Address
+                    if ((channelNumber >> (3 - i) & 0x01) != 0)
+                    {
+                        _digital.Write(_address, 1);
+                    }
+                    else
+                    {
+                        _digital.Write(_address, 0);
+                    }
+                }
+
+                // read 10-bit data
+                value <<= 1;
+                if (_digital.Read(_dataOut) == PinValue.High)
+                {
+                    value |= 0x01;
+                }

```

Comment:
```suggestion
                value |= _digital.Read(_dataOut) == PinValue.High;
```

perhaps even just `value |= _digital.Read(_dataOut);` will work too

Reply:
`Operator '|=' cannot be applied to operands of type 'int' and 'PinValue'`
It would need unnecessary cast every loop and in my opinion it would lower the readability of this part.